pkg/watcher/allocations.go

```diff
@@ -47,22 +49,44 @@ func (w *Watcher) runAllocationWatcher() {
 		log.Debug().Msg("allocations index has changed")
 
 		for _, alloc := range allocations {
-
+			// If the allocation's index hasn't changed, then there is nothing to notify about
 			if !w.indexHasChange(alloc.ModifyIndex, maxFound) {
 				log.Debug().Str("alloc-id", alloc.ID).Msg("allocation index has not changed")
 				continue
 			}
 
-			if len(alloc.TaskGroup) == 0 {
```

Comment:
For allocations, `TaskGroup` is just a string with the _name_ of the TaskGroup - I'm fairly sure that can never be empty, so it seems unnecessary to check.